### PR TITLE
Organizer certificates

### PR DESCRIPTION
**Pull request by jlauinger**
Provide a means of generating certificates for organizers.

### CODE REVIEW

File: staff/assets/orgacerts.sty [hunk 1]
```diff
@@ -35,27 +28,33 @@
...
+        \Huge \textbf{#1#2}\\
+        \vspace{0.7cm}
+        \Large für tatkräftige Hilfe bei der Organisation der Orientierungsphase\\
+        am Fachbereich Informatik der TU Darmstadt.
+    \end{center}
+
+    \vspace{3.5cm}
+
+    \begin{tabularx}{15cm}{cp{3cm}c}
+    Prof. Dr. rer. nat. Max Mühlhäuser & & Prof. Dr.-Ing. Michael Goesele\\
```
Comment (ckleemann) [inline staff/assets/orgacerts.sty]: This should be somewhere in a config object not in the code

File: staff/assets/orgacerts.sty [hunk 2]
```diff
@@ -0,0 +1,60 @@
+  % generelle Pakete
+\usepackage{graphicx}
+\usepackage{tabularx}
+\usepackage[utf8]{inputenc}
+\usepackage[german]{babel}
```
Comment (exploide) [inline staff/assets/orgacerts.sty]: probably you want to use ngerman

File: staff/templates/staff/reports/orga-report.tex [hunk 3]
```diff
@@ -3,7 +3,10 @@
 \documentclass{article}
 \usepackage{orgacerts}
 
-% Pfad zum Ophasen-Wesen
+% Pfad zu Bildern
+\newcommand{\logoFS}{logo_fs.png}
```
Comment (ckleemann) [inline staff/templates/staff/reports/orga-report.tex]: Most of the logos are already presend in the website app as svg. I think it is better to use the svg versions and to store them only once

File: staff/reports.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from django.http import HttpResponse
+from django.shortcuts import render
+
+from ophasebase.helper import LaTeX
+from ophasebase.models import Ophase
+
+
+def generate_cert(queryset):
```
Comment (exploide) [inline staff/reports.py]: generate_cert is rather unspecific. maybe name it `generate_orga_cert`

File: staff/reports.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    """ Generates a PDF file with orga certificates for staff people in the queryset """
+    current_ophase = Ophase.current()
+    (pdf, pdflatex_output) = LaTeX.render({"items": queryset, "current_ophase": current_ophase},
+                                          'staff/reports/orga-report.tex', ['orgacerts.sty', 'OPhasenWesen.png',
+                                          'logo_fs.png', 'logo_fb.png', 'logo_tu.png'],
+                                          'staff')
+    return (pdf, pdflatex_output)
+
+
+def generate_cert_response(request, queryset):
```
Comment (exploide) [inline staff/reports.py]: same here

File: staff/assets/orgacerts.sty (hunk 1 above)
Comment (jlauinger) [inline staff/assets/orgacerts.sty]: in the latest design, this table is specifically `\hspace`-adjusted. The shift value depends on the rendered text length of the names. Therefore I will leave the names hard-coded for now.

File: staff/templates/staff/reports/orga-report.tex (hunk 3 above)
Comment (jlauinger) [inline staff/templates/staff/reports/orga-report.tex]: Due to vertical alignment issues I combined all images into single wide one, so this is therefore invalid.

**Comment by jlauinger** [comment]
@ckleemann I know, but as noted above: the table is specifically `\hspace`-shifted depending on the rendered text width of the names. As that value would also have to be adjusted when the names change, I currently see no proportion in introducing a settings model for that.

**Pull request merged**